compiler/rustc_mir_build/src/builder/mod.rs

```diff
@@ -789,6 +789,28 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         builder
     }
 
+    #[allow(dead_code)]
+    fn dump_for_debugging(&self) {
+        let body = Body::new(
+            MirSource::item(self.def_id.to_def_id()),
+            self.cfg.basic_blocks.clone(),
+            self.source_scopes.clone(),
+            self.local_decls.clone(),
+            self.canonical_user_type_annotations.clone(),
+            self.arg_count.clone(),
+            self.var_debug_info.clone(),
+            self.fn_span.clone(),
+            self.coroutine.clone(),
+            None,
+        );
+        //body.coverage_info_hi = self.coverage_info.clone().map(|b| b.into_done());
```

Comment:
The coverage info builder is not Clone and I'm not sure if we should implement Clone for it anyway..

Reply:
I've added something like your PR except as a separate method to avoid extra allocations when not doing a dump.